Fails to compile on MSVC

The error  happens after: d04d8aba7029a2501854a2838fd282b81358a54e
```
6>  mapblock_mesh.cpp
6>D:\Users\Doll\Documents\Projects\Minetest\src\mapblock_mesh.cpp(1094): error C2668: 'abs': ambiguous call to overloaded function
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(328): note: could be 'long double abs(long double) noexcept'
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(30): note: or       'float abs(float) noexcept'
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(17): note: or       'double abs(double) noexcept'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\stdlib.h(363): note: or       '__int64 abs(const __int64) throw()'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\stdlib.h(358): note: or       'long abs(const long) throw()'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\math.h(465): note: or       'int abs(int)'
6>  D:\Users\Doll\Documents\Projects\Minetest\src\mapblock_mesh.cpp(1094): note: while trying to match the argument list '(irr::u32)'
6>D:\Users\Doll\Documents\Projects\Minetest\src\mapblock_mesh.cpp(1096): error C2668: 'abs': ambiguous call to overloaded function
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(328): note: could be 'long double abs(long double) noexcept'
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(30): note: or       'float abs(float) noexcept'
6>  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\cmath(17): note: or       'double abs(double) noexcept'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\stdlib.h(363): note: or       '__int64 abs(const __int64) throw()'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\stdlib.h(358): note: or       'long abs(const long) throw()'
6>  C:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt\math.h(465): note: or       'int abs(int)'
6>  D:\Users\Doll\Documents\Projects\Minetest\src\mapblock_mesh.cpp(1096): note: while trying to match the argument list '(irr::u32)'
```

--- a/src/mapblock_mesh.cpp
+++ b/src/mapblock_mesh.cpp
@@ -1090,10 +1090,10 @@ MapBlockMesh::MapBlockMesh(MeshMakeData *data, v3s16 camera_offset):
 				brightness of vertices 1 and 3 differ less than
 				the brightness of vertices 0 and 2.
 			*/
-			if (abs(f.vertices[0].Color.getAverage()
-					- f.vertices[2].Color.getAverage())
-					> abs(f.vertices[1].Color.getAverage()
-					- f.vertices[3].Color.getAverage()))
+			if (abs(f.vertices[0].Color.getLuminance()
+					- f.vertices[2].Color.getLuminance())
+					> abs(f.vertices[1].Color.getLuminance()
+					- f.vertices[3].Color.getLuminance()))
 				indices_p = indices_alternate;
 
 			collector.append(f.tile, f.vertices, 4, indices_p, 6);

--- a/src/particles.h
+++ b/src/particles.h
@@ -32,8 +32,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 struct ClientEvent;
 class ParticleManager;
 class ClientEnvironment;
-class MapNode;
-class ContentFeatures;
+struct MapNode;
+struct ContentFeatures;
 
 class Particle : public scene::ISceneNode
 {

